integr8ly/keycloak-operator
# E2e tests

## Motivation
Keycloak operator e2e tests

## What
PR contains keycloak operator-sdk framework e2e tests (still WIP)

## Why
Add an short answer for: Why it was done? (E.g The feature X was deprecated.)

## How
Add an short answer for: How it was done? (E.g By removing this feature from ... OR By removing just the button but not its implementation ... ) 

## Verification Steps

Change the image name/tag in `deploy/operator.yaml` if using a different image name.

Build the image:
```
operator-sdk build --enable-tests quay.io/integreatly/keycloak-operator:test
```

Push it to a registry:

```
docker push quay.io/integreatly/keycloak-operator:test
```

Change the image name in deploy/test-e2e.yaml if needed as well.

Add roles, role bindings and crds in your namespace:

```
oc apply -f deploy/rbac.yaml -n keycloak-test
oc apply -f deploy/Keycloak_crd.yaml -n keycloak-test
oc apply -f deploy/KeycloakRealm_crd.yaml -n keycloak-test
```

Deploy the test pod:

```
oc apply -f deploy/test-e2e.yaml -n keycloak-test
```

You can check the pod(s) logs to see how tests are going. 


## Checklist:


## Progress
